libs/api/domains/national-registry/src/lib/soap/nationalRegistryApi.ts

```diff
@@ -0,0 +1,386 @@
+import { NotFoundException } from '@nestjs/common'
+import { logger } from '@island.is/logging'
+import Soap from 'soap'
+import { MyInfo } from '../myInfo.model'
+import { GetViewHusaskraDto } from './dto/getViewHusaskraDto'
+import { GetViewKennitalaOgTrufelag } from './dto/getViewKennitalaOgTrufelag'
+import { GetViewSveitarfelagDto } from './dto/getViewSveitarfelagDto'
+import { GetViewThjodskraDto } from './dto/getViewThjodskraDto'
+import { GetViewFjolskyldanDto } from './dto/getViewFjolskyldanDto'
+import { FamilyMember } from '../familyMember.model'
+import { GetViewKennitalaOgBannmerkingDto } from './dto/getViewKennitalaOgBannmerkingDto'
+
+export class NationalRegistryApi {
+  private readonly client: Soap.Client
+  private readonly clientUser: string
+  private readonly clientPassword: string
+
+  constructor(
+    private soapClient: Soap.Client,
+    clientPassword: string,
+    clientUser: string,
+  ) {
+    this.client = soapClient
+    this.clientUser = clientUser
+    this.clientPassword = clientPassword
+  }
+
+  public async getMyInfo(nationalId: string): Promise<MyInfo | null> {
+    const response = await this.getViewThjodskra(nationalId)
+
+    if (!response)
+      throw new NotFoundException(
+        `user with nationalId ${nationalId} not found in national Registry`,
+      )
+
+    const userInfo = response.table.diffgram.DocumentElement.Thjodskra
+    const houseResponse = await this.getViewHusaskra(userInfo.LoghHusk)
```

Comment:
Is there any reason why the soap dtos are in english rather than in icelandic?

Reply:
you mean keep them all in icelandic?

Reply:
hah yeah my bad :smile: